hooks/useDragSelect.ts

```diff
@@ -0,0 +1,76 @@
+import { useState, useRef, useEffect, RefObject } from "react";
+
+interface UseDragSelectReturn {
+  draggedIndex: number[];
+}
+
+function useDragSelect<T extends HTMLElement>(
+  containerRef: RefObject<T>,
+  handleDragEnd: (draggedIndex: number[]) => void,
+): UseDragSelectReturn {
+  const [isDragging, setIsDragging] = useState(false);
+  const [startIndex, setStartIndex] = useState<number | null>(null);
+  const [endIndex, setEndIndex] = useState<number | null>(null);
+  const [draggedIndex, setDraggedIndex] = useState<number[]>([]);
+
+  useEffect(() => {
+    const handlePointerDown = (e: PointerEvent) => {
+      if (containerRef.current && containerRef.current.contains(e.target as Node)) {
+        const targetIndex = parseInt((e.target as HTMLElement).dataset.index || "-1", 10);
+        if (targetIndex !== -1) {
+          setIsDragging(true);
+          setStartIndex(targetIndex);
+          setEndIndex(targetIndex);
+          setDraggedIndex([targetIndex]);
+        }
+      }
+    };
```

Comment:
```suggestion
    const handlePointerDown = (e: PointerEvent) => {
      if (!containerRef.current || !containerRef.current.contains(e.target as Node)) {
        return;
      }

      const targetIndex = parseInt((e.target as HTMLElement).dataset.index || "-1", 10);
      if (targetIndex === -1) {
        return;
      }

      setIsDragging(true);
      setStartIndex(targetIndex);
      setEndIndex(targetIndex);
      setDraggedIndex([targetIndex]);
    };
```
p5: 이건 취향 차이인 것 같은데 저는 이렇게 `early return`을 사용하는 것을 선호합니다!